src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+    }
+
+    public static void addDeadline(String taskToAdd){
+        try {
+            validateInput(taskToAdd, "deadline");
+            int splitIndex = taskToAdd.indexOf('/');
+            String description = taskToAdd.substring(0, splitIndex);
+            String dateTime = taskToAdd.substring(splitIndex + 3);
+            Deadline d = new Deadline(description, dateTime);
+            tasks.add(d);
+            System.out.println("Got it. I've added this task:");
+            System.out.println(d.getTaskType() + d.getStatusIcon() + " " + d.getDescription() + " (by: "
+                    + d.getDateTime() + ")");
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+
+        } catch (DukeException | IndexOutOfBoundsException e) {
+            printInvalidMessage("deadline");
+        }
+    }
+
+    public static void deleteTask(String inputCommand) {
+        int taskIndex = Integer.parseInt(inputCommand) - 1;
+        if (taskIndex < 0 || taskIndex >= tasks.size()) {
+            System.out.println("Task does not exists!");
+        } else {
+            System.out.println("Noted. I've removed this task:");
+            Task currentTask = tasks.get(taskIndex);
+            System.out.println(currentTask.getTaskType() + currentTask.getStatusIcon() + " "
+                    + currentTask.getDescription());
+            tasks.remove(taskIndex);
+            if (tasks.size() > 1) {
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            } else  {
+                System.out.println("Now you have " + tasks.size() + " task in the list.");
+            }
+        }
+    }
+
+    public static void loadData() {
+        try {
+            Path dataFilePath = Paths.get("data/");
+            Files.createDirectories(dataFilePath);
+            File data = new File ("data/duke.txt");
+            Scanner readFile = new Scanner (data);
+            while (readFile.hasNextLine()) {
+                String input = readFile.nextLine();
+                //split string by whitespace + "|" + whitespace
+                String[] taskSplit = input.split("\\s\\|\\s");
+                if (taskSplit[0].equals("[D]")) {
+                    Deadline d = new Deadline (taskSplit[2], taskSplit[3]);
+                    tasks.add(d);
+                    if (taskSplit[1].equals("1")) {
+                        d.setDone();
+                    }
+                } else if (taskSplit[0].equals("[T]")) {
+                    Todo t = new Todo (taskSplit[2]);
+                    tasks.add(t);
+                    if (taskSplit[1].equals("1")) {
+                        t.setDone();
+                    }
+                } else if (taskSplit[0].equals("[E]")) {
+                    Event e = new Event (taskSplit[2], taskSplit[3]);
+                    tasks.add(e);
+                    if (taskSplit[1].equals("1")) {
+                        e.setDone();
+                    }
```

Comment:
There seem to be some common statements in each of the branches - perhaps consider extracting them to reduce the number of lines in this method - thereby improving readability.